MDHelpers/MDStatics.cs

```diff
@@ -789,5 +812,83 @@ private static IMDDataConverter _InternalGetConverterForType(Type Type)
                 return new MDObjectDataConverter();
             }
         }
+
+    /// <summary>
+    /// Find all types in the current assembly that implements the attribute
+    /// </summary>
+    /// <typeparam name="T">The attribute you want to look for</typeparam>
+    /// <returns>A list of types that implements the attribute or an empty list</returns>
+    public static List<Type> FindAllScriptsImplementingAttribute<T>() where T : Attribute
+    {
+        List<Type> List = new List<Type>();
+        Assembly assembly = Assembly.GetExecutingAssembly();
+        foreach (Type type in assembly.GetTypes())
+        {
+            if (FindClassAttribute<T>(type) != null)
+            {
+                List.Add(type);
+            }
+        }
+        return List;
+    }
+
+#region LOAD SCENES MATCHING COMPARATOR
+
+    /// <summary>
+    /// Find all PackedScenes in your project that matches the comparator.
+    /// Example: List<PackedScene> TestList = MDStatics.LoadScenes((Node n) => n.GetType().GetInterface("IAutomaticTest") != null);
+    /// This would find all scenes that implements the IAutomaticTest interface
+    /// </summary>
+    /// <param name="CompareFunc">A compare function used to check if we want this scene or not</param>
+    /// <returns>A list of PackedScenes or an empty list if none were found</returns>
+	public static List<PackedScene> LoadScenes(Func<Node, bool> CompareFunc)
+	{
+        List<PackedScene> List = new List<PackedScene>();
+		LoadScenes("res://", List, CompareFunc);
+        return List;
+	}
+
+	private static void LoadScenes(string Path, List<PackedScene> List, Func<Node, bool> CompareFunc)
+	{
+		Directory dir = new Directory();
+		dir.Open(Path);
+		dir.ListDirBegin(true, true);
+		while (true)
+		{
+			String filePath = dir.GetNext();
+			if (filePath == "")
+			{
+				break;
+			}
+			if (dir.CurrentIsDir())
+			{
+				// Go into all subfolder except ignore folder
+				LoadScenes(Path + filePath + "/", List, CompareFunc);
+			}
+			else if (filePath.ToLower().EndsWith(".tscn"))
+			{
+				PackedScene Scene = LoadPackedScene(Path + filePath);
+				Node instance = Scene.Instance();
+				if (CompareFunc(instance))
+				{
+					List.Add(Scene);
+				}
+				instance.QueueFree();
+			}
+		}
+		dir.ListDirEnd();
+	}
+
+	private static PackedScene LoadPackedScene(String path)
+	{
+		String full_path = path;
+		if (!ResourceLoader.Exists(full_path))
+		{
+			GD.Print("Can't find: " + full_path);
```

Comment:
MDLog and these changes seem to be using tabs

Reply:
Was copy / paste mostly from one of my other projects, fixed this now.